File: fiddling/static/systems/ai_system.js
import { System } from './systems.js'
import { ai_components } from '../components/ai_components.js'

class AISystem extends System {
    constructor(){
	super()
	this.components_dict = ai_components
    }
    get_component(type, entity, target){
	var component = new this.components_dict[type](entity, target)
	this.components.push(component)
	return component
    }
    update(dt){
	this.garbage_collect()
	for(var component of this.components){
	    component.update(dt)
	}
    }
}

export var ai_system = new AISystem()

File: fiddling/server/systems/ai_system.py
from .systems import System
from server.components.ai_components import ai_components


class AISystem(System):
    def __init__(self):
        super().__init__()
        self.components_dict = ai_components

    def get_component(self, entity, type, target):
        component = self.components_dict[type](entity, target)
        self.components.append(component)
        return component


ai_system = AISystem()
